lib/dry/validation/evaluator.rb

```diff
@@ -187,11 +187,17 @@ def schema_error?(path)
 
       # Check if there are any errors on the current rule
       #
+      # @param path [Symbol, String, Array] A Path-compatible spec
+      #
       # @return [Boolean]
       #
       # @api public
-      def rule_error?
-        !key(path).empty?
+      def rule_error?(path = nil)
+        if path.nil?
+          !key(self.path).empty?
+        else
+          result.error?(path)
```

Comment:
This checks all errors, not just rule errors, so I'm afraid this doesn't do what we want. One option to make this work would be to add `Result#rule_error?` that would filter out schema errors and look only at rule errors.

Reply:
The problem here that schema and rule errors are mixed in `Result#errors` set.
So I see two possible solutions:
1. Filter schema errors keys from errors set and searching in that subset.
2. Check that there are no schema error with provided key and then look up `Result#errors`.

The first approach seems to me is a little more straightforward.
Because in second one you need to know that rule dependent on schema and does not evaluate if there are some error.
But the second one was easier to implement 😄 

If second approach is not ok for you I will rewrite it.